text/0000-cfg-logical-ops.md

```diff
@@ -0,0 +1,94 @@
+- Feature Name: `cfg_logical_ops`
+- Start Date: 2025-03-30
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+`#[cfg]`, `#[cfg_attr]`, and `cfg!()` can use `&&`, `||`, and `!` for `all`, `any`, and `not`,
+respectively.
+
+# Motivation
+[motivation]: #motivation
+
+While there are no technical restrictions to using logical operators, this was not always the case.
+In Rust 1.0, attributes could not contain arbitrary tokens. This restriction was lifted in Rust
+1.34, but the `cfg` syntax was not updated to take advantage of this. By letting developers use
+logical operators, we are _lessening_ the burden of having to remember the `cfg` syntax.
+
+# Explanation
+[explanation]: #explanation
+[cfg-syntax]: https://doc.rust-lang.org/reference/conditional-compilation.html#r-cfg.syntax
+
+`#[cfg(foo && bar)]` enables the annotated code if and only if both `foo` **and** `bar` are enabled.
+Similarly, `#[cfg(foo || bar)]` enables the annotated code if and only if either `foo` **or** `bar`
+is enabled. Finally, `#[cfg(!foo)]` enables the annotated code if and only if `foo` is **not**
+enabled. `#[cfg_attr]` and `cfg!()` behave the same way. Precedence is the same as in expressions.
+
+In terms of formal syntax, the [`[cfg.syntax]`][cfg-syntax] is changed to the following:
+
+> **<sup>Syntax</sup>**\
+> _ConfigurationPredicate_ :\
+> &nbsp;&nbsp; &nbsp;&nbsp; _ConfigurationOption_\
+> &nbsp;&nbsp; | _ConfigurationAll_\
+> &nbsp;&nbsp; | _ConfigurationAny_\
+> &nbsp;&nbsp; | _ConfigurationNot_\
+> &nbsp;&nbsp; | _ConfigurationAnd_\
+> &nbsp;&nbsp; | _ConfigurationOr_\
+> &nbsp;&nbsp; | _ConfigurationNotOption_\
+> &nbsp;&nbsp; | `(` _ConfigurationPredicate_ `)`
+>
+> _ConfigurationOption_ :\
+> &nbsp;&nbsp; [IDENTIFIER]&nbsp;(`=` ([STRING_LITERAL] | [RAW_STRING_LITERAL]))<sup>?</sup>
+>
+> _ConfigurationAll_\
+> &nbsp;&nbsp; `all` `(` _ConfigurationPredicateList_<sup>?</sup> `)`
+>
+> _ConfigurationAny_\
+> &nbsp;&nbsp; `any` `(` _ConfigurationPredicateList_<sup>?</sup> `)`
+>
+> _ConfigurationNot_\
+> &nbsp;&nbsp; (`not` | `!`) `(` _ConfigurationPredicate_ `)`
+>
+> _ConfigurationAnd_\
+> &nbsp;&nbsp; _ConfigurationPredicateList_ `&&` _ConfigurationPredicateList_
+>
+> _ConfigurationOr_\
+> &nbsp;&nbsp; _ConfigurationPredicateList_ `||` _ConfigurationPredicateList_
```

Comment:
I expect `a, b && c, d` to not be valid where `a, b` is not valid, also I expect `a, b && c, d` to not mean `(a, b) && (c, d)`

```suggestion
> &nbsp;&nbsp; _ConfigurationPredicate_ `&&` _ConfigurationPredicate_
>
> _ConfigurationOr_\
> &nbsp;&nbsp; _ConfigurationPredicate_ `||` _ConfigurationPredicate_
```

Reply:
funnily enough I had this originally but misread it when copying into the final format